util/types/time.go

```diff
@@ -466,6 +471,13 @@ func (t *Time) Sub(t1 *Time) Duration {
 	}
 }
 
+// TimestampDiff returns t2 - t1 where t1 and t2 are date or datetime expressions.
+// The unit for the result (an integer) is given by the unit argument.
+// The legal values for unit are "YEAR" "QUARTER" "MONTH" "DAY" "HOUR" "SECOND" and so on.
+func TimestampDiff(unit string, t1 Time, t2 Time) int64 {
```

Comment:
Because timestampDiff in `mytime.go` is not exported, expression package need use it.

I keep most mysql time function in `mytime.go` and keep them internal, while expose some of the for outside usage.